sources/api/static-pods/src/static_pods.rs

```diff
@@ -0,0 +1,320 @@
+
+        #[snafu(display("Logger setup error: {}", source))]
+        Logger { source: log::SetLoggerError },
+
+        #[snafu(display("Unable to base64 decode manifest '{}': '{}'", base64_string, source))]
+        Base64Decode {
+            base64_string: String,
+            source: base64::DecodeError,
+        },
+
+        #[snafu(display("Failed to create directory '{}': '{}'", dir.display(), source))]
+        Mkdir {
+            dir: PathBuf,
+            source: std::io::Error,
+        },
+
+        #[snafu(display("Failed to write manifest for static pod '{}': {}", name, source))]
+        ManifestWrite {
+            name: String,
+            source: std::io::Error,
+        },
+
+        #[snafu(display(
+            "Failed to delete manifest file for static pod '{}': {}'",
+            name,
+            source
+        ))]
+        ManifestDelete {
+            name: String,
+            source: std::io::Error,
+        },
+    }
+}
+
+type Result<T> = std::result::Result<T, error::Error>;
+
+/// Query the API for the currently defined static pods
+async fn get_static_pods<P>(socket_path: P) -> Result<Option<HashMap<Identifier, model::StaticPod>>>
+where
+    P: AsRef<Path>,
+{
+    debug!("Querying the API for settings");
+
+    let method = "GET";
+    let uri = API_SETTINGS_URI;
+    let (code, response_body) = apiclient::raw_request(&socket_path, uri, method, None)
+        .await
+        .context(error::APIRequest { method, uri })?;
+    ensure!(
+        code.is_success(),
+        error::APIResponse {
+            method,
+            uri,
+            code,
+            response_body,
+        }
+    );
+
+    // Build a Settings struct from the response string
+    let settings: model::Settings =
+        serde_json::from_str(&response_body).context(error::ResponseJson { method, uri })?;
+
+    Ok(settings
+        .kubernetes
+        .context(error::MissingSettings)?
+        .static_pods)
+}
+
+/// Write out the manifest file to the pod manifest path with a given filename
+fn write_manifest_file<S1, S2>(name: S1, manifest: S2) -> Result<()>
+where
+    S1: AsRef<str>,
+    S2: AsRef<[u8]>,
+{
+    let name = name.as_ref();
+
+    let dir = Path::new(STATIC_POD_DIR);
+    fs::create_dir_all(&dir).context(error::Mkdir { dir: &dir })?;
+    let path = dir.join(name);
+    // Create the file if it does not exist, completely replace its contents
+    // if it does (constitutes an update).
+    fs::write(path, manifest).context(error::ManifestWrite { name })?;
+
+    Ok(())
+}
+
+/// Deletes the named manifest file if it exists
+fn delete_manifest_file<S1>(name: S1) -> Result<()>
+where
+    S1: AsRef<str>,
+{
+    let name = name.as_ref();
+    let path = Path::new(STATIC_POD_DIR).join(name);
+    if path.exists() {
+        fs::remove_file(path).context(error::ManifestDelete { name })?;
+    }
+
+    Ok(())
+}
+
+/// Store the args we receive on the command line
+struct Args {
```

Comment:
nit: these arg parsing functions seem to split up the main logic functions of the program.  `handle_static_pod` is all by their lonesome below!  I might consider grouping the arg parsing and other logic functions together.